### ISSUE TITLE
Support for InputCapture and RemoteDesktop Wayland portals

### ISSUE THREAD

**Issue by dmac-au** [pop-os/cosmic-comp#980]
I installed [Deskflow](https://github.com/deskflow/deskflow) (Synergy open source app) and noticed an error in Cosmic on wayland:

```shell
[2024-11-08T04:11:34] ERROR: failed to initialize input capture session, quitting: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.InputCapture” on object at path /org/freedesktop/portal/desktop
```

I thought I could then instead use the Windows PC as the host and the Linux as the guest, but then I got this error:

```shell
[2024-11-08T04:23:12] ERROR: failed to initialize remote desktop session: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.RemoteDesktop”
```

Can we add InputCapture and RemoteDesktop portals to the roadmap if they aren't already?

I probably have to switch back to KDE for the time being (I haven't tested it with Deskflow yet, they will get the same bug report if it doesn't work)

**Comment by ids1024** [comment]
A partly complete prototype for the `RemoteDesktop` portal exists, but isn't merged: https://github.com/pop-os/xdg-desktop-portal-cosmic/pull/94, https://github.com/pop-os/cosmic-comp/pull/465.

There hasn't been any work specifically on `InputCapture` so far.

**Comment by Drakulix** [comment]
Also this isn't a compositor issue, but rather a portal issue.

**Comment by ids1024** [comment]
Yeah, though both features require underlying support from the compositor.

I think we also have several issues on multiple repositories related to `RemoteDesktop`/synergy-like uses at this point.

**Comment by dmac-au** [comment]
I found this issue in the portals repo. I'll close this here.
